### PR TITLE
🚑 Update Invitation Routes

### PR DESCRIPTION
**Pull request by shnizzedy**
# Changes
* Updated decline route
  * from `DELETE /invitation/{id}/remove` to just `DELETE /invitation/{id}`
  * any logged in user can now decline (delete) an invitation
* Updated invitation HTML
  * changed link from acceptance to `GET` call to link to invitation on ML-web
  * added `includeLink` param so that link can be excluded when rendered on the web
* Marked `GET user/invites` as deprecated (`GET /applet/{id}/users` includes all open invitations in `pending`)
* Fixed an order-of-operations bug that was causing `POST /invitation/{id}/accept` to fail

Related https://github.com/ChildMindInstitute/MindLogger-bug-reports/issues/43

### CODE REVIEW

**Comment by shnizzedy** [comment]
@henryrossiter I just deployed 68fb8d466723bc1f8c527db8dfad3411bca55c4a (with the changes above) to dev, but I haven't figured out the `400` response yet. Working on it now.

**Comment by shnizzedy** [comment]
7820cb4 is live on dev, and I think each of these routes should work now.

File: girderformindlogger/api/v1/invitation.py [hunk 1]
```diff
@@ -196,8 +195,5 @@ def declineInvitation(self, invitation):
             raise AccessException(
                 "You must be logged in to accept an invitation."
```
Comment (henryrossiter) [inline girderformindlogger/api/v1/invitation.py]: Small typo here, should say decline or remove instead of accept

Comment (shnizzedy) [reply girderformindlogger/api/v1/invitation.py]: @henryrossiter good catch! https://github.com/ChildMindInstitute/mindlogger-app-backend/pull/227/commits/b18885b864140aea4a2abc70b2fcbe86b9a06cd2 gets rid of the log in requirement for declining. I think we don't want to force people to create an account just to decline, right?

Comment (henryrossiter) [reply girderformindlogger/api/v1/invitation.py]: That works as well. 

I'm not sure I understand what the delete call does in the first place though. Even after a user declines an invitation, he/she could still view the invitation by clicking the same link, correct? Then he/she could choose whether to accept or decline again?

Comment (shnizzedy) [reply girderformindlogger/api/v1/invitation.py]: Nope. The delete call deletes the invitation.

**Pull request merged**